ecs-init/cache/cache_test.go

```diff
@@ -256,10 +272,13 @@ func TestDownloadAgentCopyFailure(t *testing.T) {
 	mockFS.EXPECT().TeeReader(tarballResponse.Body, gomock.Any())
```

Comment:
Same as above. Can we be more strict and the check the ordering here (Using [`gomock.InOrder`](https://godoc.org/github.com/golang/mock/gomock#InOrder))?